Add Yew

jetli opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

**https://github.com/jetli/awesome-yew**

**A curated list of awesome things related to Yew / WebAssembly. Yew is a Rust / Wasm framework for building client web apps, top 1 frontend web framework in Rust, with 10k+ stars. Yew is inspired by Elm and React for creating multi-threaded frontend apps with WebAssembly. This list should help frontend developers with learning Yew / WebAssembly architectural patterns and with creating fast and reliable WASM web apps.**

PRs reviewed:
https://github.com/sindresorhus/awesome/pull/1684#issuecomment-576959071
https://github.com/sindresorhus/awesome/pull/1687#issuecomment-576961333
https://github.com/sindresorhus/awesome/pull/1622#issuecomment-576962009

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).**
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- You have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

jetli commented:
unicorn

jetli commented:
PRs reviewed:
https://github.com/sindresorhus/awesome/pull/1684#issuecomment-576959071

agucova commented:
Awesome list! I loved the logo.

However, the license should not be included in the README, per:

> Do not add the license name or text to the readme. GitHub already shows the license name at the top of the repo.

Beyond that, the list seems to pass all the requirements with flying colors

jetli commented:
@agucova  Done, removed the redundant License section: https://github.com/jetli/awesome-yew/commit/d2a69d9c6942056a3a30ca393c32c100414c56b1. Thanks!

sindresorhus commented:
The intro text in your Awesome list could be a little bit more verbose than just `Yew is a Rust / Wasm framework for building client web apps.`.

---

The header image is nice, but it's weird having the text "Awesome" twice.  I think you should add "Yew" below the "Awesome" text in the header image (in the same font). Then you can drop the text heading.

sindresorhus commented on readme.md:
```diff
@@ -296,6 +296,7 @@
 - [Seed](https://github.com/seed-rs/awesome-seed-rs#readme) - Rust framework for creating web apps running in WebAssembly.
 - [Web Performance Budget](https://github.com/pajaydev/awesome-performance-budget#readme) - Techniques to ensure certain performance metrics for a website.
 - [Web Animation](https://github.com/sergey-pimenov/awesome-web-animation#readme) - Animations in the browser with JavaScript, CSS, SVG, etc.
+- [Yew](https://github.com/jetli/awesome-yew#readme) - Rust / Wasm framework for building client web apps.
```
```suggestion
- [Yew](https://github.com/jetli/awesome-yew#readme) - Rust / Wasm framework for building client-side web apps.
```

jetli commented:
@sindresorhus I've updated:

- Update the intro text to more detailed official intro of Yew:  `Rust / Wasm framework inspired by Elm and React for creating multi-threaded frontend web apps with WebAssembly.`
- For the logo part, I enlarged the size of the logo a little bit to emphasize it, and adjusted layout of the following text a bit,  so people won't easily notice two same awesome text. Because I'm new to svg editing,  it's hard for me to add more text to the logo with specific font, sorry for that, or I might try that later.

palmskog commented:
I couldn't get this to pass in `awesome-lint` without a lot of errors, and it looks like this is due to the headings not being at the right level, that is:
```
## Awesome Yew
```
should be
```
# Awesome Yew
```
and so on below, e.g., `### Templates` should be `## Templates`. When this is done, `awesome-lint` reports no errors.

I also wonder about the `Contribute` section - it is not mentioned in `Contents`, maybe because this is not allowed by `awesome-lint`? Having an orphan section like this is hardly ideal even if not explicitly forbidden, it may be better to mention contribution guidelines in some other section or not at all.

jetli commented:
@palmskog Fixed by this PR: https://github.com/jetli/awesome-yew/pull/4, thanks.

palmskog commented:
@jetli that PR indeed fixes all previous `awesome-lint` errors, but introduces the following error:
```
ToC should not contain section "Contribute"
```
... which I talked about in my original comment. The way to solve this is to move the mention of the contribution guidelines to some other section, or skip mentioning them altogether.

jetli commented:
@palmskog Removed contribute section entirely: https://github.com/jetli/awesome-yew/commit/4402d9406d8b5c06f74b9e30e2abf4a3bbb906d1, thanks.

palmskog commented:
@jetli all the formatting/structure/layout now looks good to me, and `awesome-lint` says all OK. As a final comment from my side, please take a look at consistency of names and abbreviations:
- `rust` vs. `Rust`
- `WebAssembly` vs. `wasm` vs. `Wasm`
- `html` vs. `HTML`

jetli commented:
@palmskog Standardized to Rust/WebAssembly/HTML, and added a new Related lists section. Thanks.

jetli commented:
Fixed the header image by moving to the right side and saved a lot of space for the real contents.

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1229036734165307392
